lib/holding_loader.rb

```diff
@@ -5,16 +5,24 @@
 
 # Constructs batches of Holdings from incoming file data
 class HoldingLoader
+
+  def self.get_for(filename)
+    if filename.end_with?(".tsv")
+      HoldingLoaderTsv.new
+    elsif filename.end_with?(".ndj")
+      HoldingLoaderNdj.new
+    else
+      raise "given an invalid file extension"
+    end
+  end
+
   def initialize
     @organization = nil
     @current_date = nil
   end
 
-  def item_from_line(line)
-    Holding.new_from_holding_file_line(line).tap do |h|
-      @organization ||= h.organization
-      @current_date ||= h.date_received
-    end
+  def item_from_line(_line)
```

Comment:
I did this in order to screw with the class as little as possible. 
I'm not sure I understand your point, so I'll leave it as is unless you really think it's worth doing.

Reply:
Generally I try to avoid abstract superclasses in Ruby, but it's not a big issue right now. If we find later that we're having trouble testing `HoldingsLoader` it might be worth considering doing the refactoring.